pkg/source/reconciler/source/config_watcher.go

```diff
@@ -0,0 +1,160 @@
+package source
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/eventing-kafka/pkg/common/constants"
+	"knative.dev/eventing/pkg/reconciler/source"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	EnvKafkaCfg           = "K_KAFKA_CONFIG"
+	kafkaConfigMapNameEnv = "CONFIG_KAFKA_NAME"
+)
+
+type KafkaConfig struct {
+	SaramaYamlString string
+}
+
+type KafkaSourceConfigAccessor interface {
+	source.ConfigAccessor
+	KafkaConfig() *KafkaConfig
+}
+
+// ToEnvVars serializes the contents of the ConfigWatcher to individual
+// environment variables.
+func (cw *KafkaSourceConfigWatcher) ToEnvVars() []corev1.EnvVar {
+	envs := cw.ConfigWatcher.ToEnvVars()
+
+	envs = maybeAppendEnvVar(envs, cw.kafkaConfigEnvVar(), cw.KafkaConfig() != nil)
+
+	return envs
+}
+
+// maybeAppendEnvVar appends an EnvVar only if the condition boolean is true.
+func maybeAppendEnvVar(envs []corev1.EnvVar, env corev1.EnvVar, cond bool) []corev1.EnvVar {
+	if !cond {
+		return envs
+	}
+
+	return append(envs, env)
+}
```

Comment:
Are there more usages planned for this function?  As it is it looks like an overly complicated way of doing this (though nothing technically amiss)
```suggestion
	if cw.KafkaConfig() != nil {
		envs = append(envs, cw.kafkaConfigEnvVar())
	}
	return envs
}
```

Reply:
Frankly, it is a copy paste of a private function in eventing. Let me see if I can make that one in eventing public.

Reply:
Hmm, I will check that after this PR as a follow up.

Reply:
https://github.com/knative-sandbox/eventing-kafka/pull/392